x/group/server/invariants.go

```diff
@@ -0,0 +1,129 @@
+package server
+
+import (
+	"fmt"
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+func (s serverImpl) RegisterInvariants(ir sdk.InvariantRegistry) {
+	ir.RegisterRoute(group.ModuleName, "Tally-Votes", s.tallyVotesInvariant())
+}
+
+func (s serverImpl) AllInvariants() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		return s.tallyVotesInvariant()(ctx)
+	}
+}
+
+func (s serverImpl) tallyVotesInvariant() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		if ctx.BlockHeight()-1 < 0 {
+			return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", "Not enough blocks to perform TallyVotesInvariant"), false
+		}
+		prevCtx, _ := ctx.CacheContext()
+		prevCtx = prevCtx.WithBlockHeight(ctx.BlockHeight() - 1)
+		msg, broken, err := tallyVotesInvariant(ctx, prevCtx, s.proposalTable)
+		if err != nil {
+			panic(err)
+		}
+		return sdk.FormatInvariant(group.ModuleName, "Tally-Votes", msg), broken
+	}
+}
+
+func tallyVotesInvariant(ctx sdk.Context, prevCtx sdk.Context, proposalTable orm.AutoUInt64Table) (string, bool, error) {
+
+	var msg string
+	var broken bool
+
+	prevIt, err := proposalTable.PrefixScan(prevCtx, 1, math.MaxUint64)
+	if err != nil {
+		msg = err.Error()
+		broken = true
```

Comment:
not sure if that makes sense to set the invariants `broken` to `true` since we couldn't even verify whether or not it was broken because of the error, additionally it will just cause a panic.
Also you're already returning the `err` in itself so no need to add it as part of the returned `msg` too

Reply:
okay